src/player/Player.php

```diff
@@ -485,6 +487,11 @@ public function setViewDistance(int $distance) : void{
 
 		$this->getNetworkSession()->syncViewAreaRadius($this->viewDistance);
 
+		if($this->viewDistance !== $oldViewDistance){
+			$ev = new PlayerViewDistanceChangeEvent($this, $this->viewDistance, $oldViewDistance);
+			$ev->call();
+		}
```

Comment:
I think this should be called before the view distance change, rather than after. If we decide to make it cancellable later on, we don't want to subtly break plugin assumptions about `Player->getViewDistance()` during the event.